core/trino-spi/src/main/java/io/trino/spi/connector/SourcePage.java

```diff
@@ -66,6 +66,12 @@ static SourcePage create(Page page)
      */
     long getRetainedSizeInBytes();
 
+    /**
+     * Returns the retained size of the {@link SourcePage} container and associated fields without including
+     * the size of any field blocks (loaded or unloaded).
+     */
+    long retainedShallowSizeInBytes();
```

Comment:
nit: shallowRetainedSizeInBytes ;-)

Reply:
I'll wait for everyone to weigh in with their favorite alternative method name before I go update all of the commits to match 😛

Reply:
Renamed